lib/trino-hive-formats/src/test/java/io/trino/hive/formats/avro/TestAirCompressorAsAvroCodec.java

```diff
@@ -0,0 +1,240 @@
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.zip.CRC32;
+
+import static io.trino.hive.formats.avro.AirCompressorCodec.SNAPPY_CODEC_SUPPLIER;
+import static io.trino.hive.formats.avro.AirCompressorCodec.ZSTD_CODEC_SUPPLIER;
+import static org.apache.avro.file.BZip2Codec.DEFAULT_BUFFER_SIZE;
+import static org.apache.avro.file.CodecFactory.DEFAULT_ZSTANDARD_LEVEL;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public class TestAirCompressorAsAvroCodec
+{
+    @DataProvider(name = "kinds")
+    public static Object[][] compressionKinds()
+    {
+        return Arrays.stream(AvroCompressionKind.values()).map(k -> new AvroCompressionKind[] {k}).toArray(Object[][]::new);
+    }
+
+    @Test(dataProvider = "kinds")
+    public void testAirCompressorCompatibility(AvroCompressionKind kind)
+    {
+        Supplier<byte[]> bytes = randomOriginalBytes();
+        Function<ByteBuffer, ByteBuffer> trinoCodec = getTrinoAvroCompression(kind);
+        Function<ByteBuffer, ByteBuffer> avroCompression = getCopiedAvroCompression(kind);
+        Function<ByteBuffer, ByteBuffer> trinoDecompress = getTrinoAvroDecompression(kind);
+        Function<ByteBuffer, ByteBuffer> avroDecompress = getCopiedAvroDecompression(kind);
+
+        ByteBuffer expected = ByteBuffer.wrap(bytes.get());
+        ByteBuffer trinoDecompressedFromAvroCompressed = trinoDecompress.apply(avroCompression.apply(ByteBuffer.wrap(bytes.get())));
+        ByteBuffer avroDecompressedFromTrinoCompressed = avroDecompress.apply(trinoCodec.apply(ByteBuffer.wrap(bytes.get())));
+        ByteBuffer trinoRoundTrip = trinoDecompress.apply(trinoCodec.apply(ByteBuffer.wrap(bytes.get())));
+        ByteBuffer avroRoundTrip = avroDecompress.apply(avroCompression.apply(ByteBuffer.wrap(bytes.get())));
+
+        assertThat(trinoDecompressedFromAvroCompressed).isEqualTo(expected);
+        assertThat(avroDecompressedFromTrinoCompressed).isEqualTo(expected);
+        assertThat(trinoRoundTrip).isEqualTo(expected);
+        assertThat(avroRoundTrip).isEqualTo(expected);
+    }
+
+    private Supplier<byte[]> randomOriginalBytes()
+    {
+        byte[] b = generateRandomByteArray();
+        return () -> Arrays.copyOf(b, b.length);
+    }
+
+    private byte[] generateRandomByteArray()
+    {
+        int n = ThreadLocalRandom.current().nextInt(1000, 10000);
+        byte[] a = new byte[n];
+        ThreadLocalRandom.current().nextBytes(a);
+        return a;
+    }
+
+    //Copied functions from Avro Codec's constructed with default params
+    static Function<ByteBuffer, ByteBuffer> getCopiedAvroCompression(AvroCompressionKind compressionKind)
```

Comment:
Shouldn't this use `getLegacyCodecFactory`? Seems like we should test against the exact Avro code.

Reply:
yeah, problem is that all Codec's have private constructors and the CodecFactory createInstance method is only usable inside the avro library. Mades it impossible to call these methods on arbitrary byte arrays,

Reply:
You can create a local utility method in the `org.apache.avro.file` which would allow you to call it:
```java
package org.apache.avro.file;

public final class CodecFactoryAccessor
{
    private CodecFactoryAccessor() {}

    public static Codec createCodecInstance(CodecFactory factory)
    {
        return factory.createInstance();
    }
}

Reply:
oh wow. Let me try

Reply:
yeah this worked. Cleaning up a bunch